api/openapi.yml

```diff
@@ -173,6 +201,27 @@ components:
         created_at:
           type: string
           format: date-time
+    ErrorResponse:
+      type: object
+      properties:
+        status:
```

Comment:
What's the purpose of sending `status`? It seems like this information is already conveyed by the HTTP status code.

Reply:
Yes it is. The status property in the JSON is useful so that error handling can be done with only the actual response data. This means that API functions in the Javascript can just return the response JSON and the caller of that API function doesn't have to be aware of the response object (which is where the status code lives).

This makes it easy to have this kind of pattern (pseudocode):
```javascript
// API file
function getQuestion(question_id) {
 let response = fetch(question_id);
 if(response.status_code == 401)
   // Handle unauthorized explicitly, probably by redirecting to login
return response.json();
}
// UI file
let response = await getQuestion(question_id);
if(response.status == 'error')
  // Handle any unknown errors 
else
  // Use response directly

Reply:
In your example, could `getQuestion` add that status field to its output? Something like:
```
return {
    ...response.json(),
    status: response.status_code
};
```
That way robokache need only be concerned with its interface being sensible, and the client implementation details are contained in the API file.

Reply:
Yeah I can do that. I still think that robokache should always return JSON responses to make it easy to consistently parse. In this case, if we decided not to return the status and error fields we would be returning an empty object on error responses. I think that's ok to do, but I just wanted to check.

Reply:
Yeah, I'm all for returning JSON. And we should still include the `message` field for errors, right?

Reply:
Yeah that's a good point. Maybe we should be returning a wrapped data object if we want to do error handling based on the json (something like `{ "status" : "ok", "data" : {} }`), but that's definitely a discussion for another time.